crypto/src/asset.rs

```diff
@@ -28,6 +32,21 @@ use crate::Fq;
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub struct Id(pub Fq);
 
+/// Validator-specific unique asset identifier, used for identifying staking tokens
+/// associated with a particular validator.
+pub struct ValidatorAssetId(pub Fq);
```

Comment:
In order to record the delegation tokens with the existing transaction model, we need to have an ordinary `asset::Id`, so rather than building a custom validator-specific ID type, I think it would be better to have a validator-specific `ValidatorDenom` type that would have an ordinary `asset::Id`.

The validator-specific denomination type could be a parsed representation, storing the validator's identity internally with methods to produce the corresponding `Id` and `Denom`, as well as the identity. It would probably also be useful to have some way to try to parse a `Denom` to find out whether or not it's a delegation token.

It might be worth thinking through this design with https://github.com/penumbra-zone/penumbra/pull/241#issuecomment-989558425 and https://github.com/penumbra-zone/penumbra/issues/244

Reply:
I took a second pass at this by implementing a new `ValidatorAsset` type with `denom -> Denom` and `id -> Id` methods, and added a `is_validator_denom(Denom) -> bool` function as well.